Implement restart from reference configuration (#224)
precice/openfoam-adapter: docs/config.md

Use the option `namePointDisplacement unused;` for solvers that do not create a pointDisplacement field, such as the RBFMeshMotionSolver.

#### Restarting FSI simulations

> Restarting a coupled simulation using the OpenFOAM adapter works in principle in the same way as restarting a standalone OpenFOAM solver. However, the adapter and preCICE always redefine the coupling interface and the interface node locations. In case of FSI simulations, the interface deforms over time, which leads to the definition of a deformed interface during the restart. The boolean variable `resetDisplacement` (`false` by default) allows to account for the previously accumulated interface deformation such that the initial interface configuration (`t = 0`) is completely recovered. The setting here needs to coincide with the behavior of the selected solid solver.

Reviewer: > However, the adapter and preCICE always redefine the coupling interface and the interface node locations.

What does "always" and "redefine" mean in this context? I am confused by this sentence.

> The setting here needs to coincide with the behavior of the selected solid solver.

I would use "agree" here instead of "coincide" (simpler word).

Author: Updated.

```c++
FSI
{
    // Account for previous displacement during a restart
    resetDisplacement true;
}
```

{% note %}

> The CalculiX adapter restarts simulations from the deformed interface mesh.

Author: added something

Note that the adapter does not automatically adapt the pressure name for solvers that account for [hydrostatic pressure effects](https://www.openfoam.com/documentation/guides/latest/doc/guide-applications-solvers-variable-transform-p-rgh.html). In these cases, you may want to set `nameP p_rgh` to couple `p_rgh`, as `p` is a derived quantity for these solvers.

#### Restarting FSI simulations

Restarting a coupled simulation using the OpenFOAM adapter works in principle in the same way as restarting a standalone OpenFOAM solver. However, the adapter and preCICE define the coupling interface and the interface node locations based on the mesh at the particular time. In case of FSI simulations, the interface deforms over time, which leads to the definition of a deformed interface during the restart. The boolean variable `restartFromDeformed` (`true` by default) allows to account for the previously accumulated interface deformation such that the initial interface configuration (`t = 0`) is completely recovered. The setting here needs to agree with the behavior of the selected solid solver.

```c++

> FSI
> {
>     // Account for previous displacement during a restart
>     restartFromDeformed true;
> }

Reviewer: The current documentation confused me for a while:

```suggestion
Interface1
{
    mesh                Fluid-Mesh;
    patches             (flap);
    
    // Account for previous displacement during a restart
    restartFromDeformed true;
}
```

Author: I moved the parameter now into the FSI dict (although it would fir conceptually better to the interface). I guess for the user, the FSI dict is the way to go.

```

{% note %}

> The CalculiX adapter restarts simulations from the deformed interface mesh, so that the default setting `restoreUndeformedInterface false` should be used.

Reviewer: This sounds to me clearer and closer to the intended style: https://precice.org/docs-meta-overview.html#language--style

```suggestion
The CalculiX adapter restarts simulations from the deformed interface mesh. This means that you need to set `restoreUndeformedInterface false`.
```

}
```

{% important %}

> The option here defines the way the interface mesh is initialized when restarting an FSI simulation in OpenFOAM. In order to restart a coupled simulation, your solid solver needs to be capable of restarting mechanism as well. You can find more information about restarting coupled simulations on [Dsicourse](https://precice.discourse.group/t/how-can-i-restart-a-coupled-simulation/675).

Reviewer: ```suggestion
The option here defines the way the interface mesh is initialized when restarting an FSI simulation in OpenFOAM. In order to restart a coupled simulation, your solid solver needs to be capable of restarting as well. Furthermore, the two participants need to follow the same assumption for the initialization, which for OpenFOAM you can configure with this option. You can find more information about restarting coupled simulations on [Dsicourse](https://precice.discourse.group/t/how-can-i-restart-a-coupled-simulation/675).
```
